src/components/ProgressBar/index.js

```diff
@@ -16,7 +16,19 @@ const ProgressBar = React.createClass({
 
   componentDidMount () {
     this.initProgressBar();
-    this.updateProgressBar(this.props.percentage);
+    if (this.props.percentage > 0) {
+      this.updateProgressBar(this.props.percentage);
+    }
+  },
+
+  componentDidUpdate (prevProps) {
+    if (prevProps.percentage !== this.props.percentage) {
+      this.updateProgressBar(this.props.percentage);
+    }
+  },
+
+  componentWillUnmount () {
+    this.progressBar.destroy();
```

Comment:
ping @artibella - all this logic is duplicated. would be nice to abstract a generic `ProgressElement` with options for circle, line, size etc. so we don't have this  in two places.